koma-core-api/common/src/koma/extensions/matrix_cross.kt

```diff
@@ -56,3 +56,69 @@ operator fun Int.times(other: Matrix<Double>) = other.times(this.toDouble())
  * Multiply a scalar by a matrix
  */
 operator fun Matrix<Double>.times(other: Int) = this * other.toDouble()
+
+/**
+ * Convert an `Iterable<T>` to a `Matrix<Double>`
+ */
+fun <T, N: Number> Iterable<T>.toMatrix(vararg selectors: (T) -> N): Matrix<kotlin.Double> {
+
+    val items = toList()
+
+    return koma.fill(items.size, selectors.size) { row, col -> selectors[col](items[row]).toDouble() }
+}
+
+/**
+ * Convert a `Iterable<T>` to a single column `Matrix<Double>`
+ */
+fun <T, N: Number> Iterable<T>.toVector(selector: (T) -> N): Matrix<kotlin.Double> {
+
+    val items = toList()
+
+    return koma.fill(items.size, 1) { row, _ -> selector(items[row]).toDouble() }
+}
+
+
+/**
+ * Convert a `Sequence<T>` to a `Matrix<Double>`
+ */
+fun <T, N: Number> Sequence<T>.toMatrix(vararg selectors: (T) -> N): Matrix<kotlin.Double> = toList().toMatrix(*selectors)
+
+@Suppress("ReplaceGetOrSet")
+        /**
+ * A helper function that allows for quick construction of matrix literals.
+ *
+ * For example, one can write
+ * ```
+ * var a = matrixOf[1,2,3 end
+ *                  4,5,6]
+ * ```
+ *
+ * to get a 2x3 [Matrix<Double>] with the given values. `end` is a helper object that indicates the end of a row
+ * to this object. Note that one currently cannot use this function to generate a column vector:
+ *
+ * ```// ERROR:```
+ *
+ * ```matrixOf[1 end 2 end 3]```
+ *
+ * Instead do this:
+ *
+ * ```// Define a column vector by transposing a row-vector```
+ *
+ * ```matrixOf[1, 2, 3].T```
```

Comment:
I think these docstrings need to be written with parentheses instead of braces, i.e. `matrixOf(1,2,3)`, as the implementation of `matrixOf` is using parens to match Kotlin's `listOf`.

Reply:
Oops. K I'll fix